src/inference/elbo.ad.js

```diff
@@ -354,8 +354,11 @@ module.exports = function(env) {
         // Signal to mapDataFinal that the batch was empty.
         this.mapDataStack.push(null);
       }
-
-      return ix;
+      if (ix === null) {
+          return data;
+      }
+      // TODO - change to a more efficient solution?
+      return data.filter(function(val, index){ return ix.includes[index] });
```

Comment:
This should be `includes(index)` rather than `includes[index]`. I realize this will change anyway, but I found it necessary to make this change when testing the code from [this comment](https://github.com/probmods/webppl/pull/697#discussion_r87580260).